build/openshift-ansible-extras/cluster-api-prep/deploy-cluster-api.yaml

```diff
@@ -0,0 +1,62 @@
+---
+- name: Setup the master node group
+  hosts: localhost
+  tasks:
+  - import_role:
+      name: openshift_aws
+      tasks_from: setup_master_group.yml
+
+#- name: set the master facts for hostname to elb
+#  hosts: masters
+#  gather_facts: no
+#  remote_user: root
+#  tasks:
+#  - import_role:
+#      name: openshift_aws
+#      tasks_from: master_facts.yml
+
+#- name: run the init
+#  import_playbook: ../init/main.yml
+
+- name: install cluster-api
+  hosts: masters[1]
+  gather_facts: no
+  remote_user: root
+  vars:
+    cluster_api_namespace: kube-cluster
+    cluster_api_image: DEFAULT_LOCATION_HERE
+  tasks:
+  - name: import lib_openshift
+    import_role:
+      name: lib_openshift
+
+  - name: slurp local template content
+    slurp:
+      src: files/cluster-api-template.yaml
+    delegate_to: localhost
+    register: out
+
+  - name: "create {{ cluster_api_namespace }} namesapce"
+    oc_project:
+      state: present
+      name: "{{ cluster_api_namespace }}"
+      display_name: "Cluster API"
+
+  - name: process template
+    oc_process:
+      namespace: "{{ cluster_api_namespace }}"
+      state: present
+      create: true
```

Comment:
I think it would be more robust to oc_apply the result of oc_process instead of passing create: true here.

Reply:
Yeah oc apply when possible is great, although is there such a module with lib_openshift? I've been resorting to CLI to do it.

Reply:
Sorry yes, there's a couple of roles that include oc_apply tasks, but all they do is end up calling the command itself.

Reply:
i tested repeatedly re-runing the oc_process, and it seems to do the right thing and not error out